diesel/src/lib.rs

```diff
@@ -1,16 +1,95 @@
 //! # Diesel
 //!
 //! Diesel is an ORM and query builder designed to reduce the boilerplate for database interactions.
+//! If this is your first time reading this documentation,
+//! we recommend you start with the [getting started guide].
+//! We also have [many other long form guides].
 //!
-//! * [Homepage](https://diesel.rs/)
-//! * [Getting Started Guide](https://diesel.rs/guides/getting-started/)
-//! * [Source Code](https://github.com/diesel-rs/diesel/)
+//! [getting started guide]: https://diesel.rs/guides/getting-started/
+//! [many other long form guides]: https://diesel.rs/guides
 //!
-//! ## Crates
+//! # Where to find things
 //!
-//! * **Diesel**: core libary, documented here
-//! * [**Diesel CLI**](https://github.com/diesel-rs/diesel/tree/master/diesel_cli): command line tool that aids in managing your database schema
+//! ## Declaring your schema
 //!
+//! Diesel ensures that your queries are valid at compile time.
+//! This means it needs to know what your schema is.
+//! This is done using [the `table!` macro][`table!`].
+//! This macro is usually generated by `diesel print-schema` or [`infer_schema!`]
+//!
+//! [`table!`]: macro.table.html
+//! [`infer_schema!`]: FIXME: Host this crate on docs.diesel.rs because docs.rs won't render docs for this crate
+//!
+//! ## Getting started
+//!
+//! Queries usually start from either a table, or a function like [`update`].
+//! Those functions can be found [here](#functions).
+//!
+//! Diesel provides a [`prelude` module](prelude),
+//! which exports most of the most important traits and types.
+//! We are conservative about what goes in this module,
+//! and avoid anything which has a generic name.
+//! Files which use Diesel are expected to have `use diesel::prelude::*;`.
```

Comment:
> are expected to

What expects this? Our derives? Macros? Maybe add this as a reason

Reply:
It's just convention. All our docs are written assuming that line. All our macros and derives should work without it.